control-plane/src/main.rs

```diff
@@ -38,12 +38,14 @@ async fn main() -> Result<()> {
         configuration::get_cert_provisoner_host(),
         3000,
         shared::ENCLAVE_CERT_PORT,
+        true,
```

Comment:
this feels a bit too easy to change accidentally/without thinking. Wdyt about having two constructors? `with_internal_dns` and `with_external_dns`?

The ideal might be to have types for the two DNS services, and have constructors bound to internal/external.

Reply:
Not sure creating the DNS service in the constructor makes great sense when there's two options.